sonic_platform_base/sonic_thermal_control/thermal_condition_base.py

```diff
@@ -0,0 +1,70 @@
+class ThermalConditionBase(object):
+    """
+    Base class for thermal condition
+    """
+    # JSON field definition
+    JSON_FIELD_CONDITION_TYPE = 'type'
+
+    # Dictionary of ThermalConditionBase-derived class representing all thermal condition types.
+    _condition_type_dict = {}
+
+    def is_match(self, thermal_info_dict):
+        """
+        Indicate if this condition is matched.
+        :param thermal_info_dict: A dictionary stores all thermal information.
+        :return: True if condition matched else False.
+        """
+        raise NotImplementedError
+
+    def load_from_json(self, json_obj):
+        """
+        Initialize this object by a json object. The json object is read from policy.json section 'conditions'.
+        Derived class can define any field in policy.json and interpret them in this function.
+        :param json_obj: A json object representing a condition.
+        :return:
+        """
+        pass
+
+    @classmethod
+    def get_type(cls, json_obj):
```

Comment:
We can add a common base class to hold get_type, register_type, load_from_json...